examples/hello-world/main.cpp

```diff
@@ -230,12 +229,12 @@ Slang::Result initialize()
 {
     // Create a window for our application to render into.
     //
-    WindowDesc windowDesc;
+    platform::WindowDesc windowDesc;
     windowDesc.title = "Hello, World!";
     windowDesc.width = gWindowWidth;
     windowDesc.height = gWindowHeight;
-    gWindow = createWindow(windowDesc);
-
+    gWindow = platform::Application::createWindow(windowDesc);
+    gWindow->events.mainLoop = [this]() { renderFrame(); };
```

Comment:
I'd kind of rather see the event-handling stuff handled by having a "delegate" sort of COM interface that handles events for the application/window/whatever. Using fine-grained event-handler callbacks like this seems like overkill.

(I guess it also weird me out to be attaching the handlers after creation, rather than as part of creation)